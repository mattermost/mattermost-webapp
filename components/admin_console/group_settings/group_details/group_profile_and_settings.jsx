// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import PropTypes from 'prop-types';
import {FormattedMessage} from 'react-intl';

import FormattedMarkdownMessage from 'components/formatted_markdown_message.jsx';
import AdminPanel from 'components/widgets/admin_console/admin_panel';
import GroupProfile from 'components/admin_console/group_settings/group_details/group_profile';

import {t} from 'utils/i18n';

import LineSwitch from 'components/admin_console/team_channel_settings/line_switch.jsx';

const GroupSettingsToggle = ({isDefault, allowReference, onToggle, isDisabled}) => (
    <LineSwitch
        id={'allowReferenceSwitch'}
        disabled={isDisabled || isDefault}
        toggled={allowReference}
        last={true}
        onToggle={() => {
            if (isDefault) {
                return;
            }
            onToggle(!allowReference);
        }}
        singleLine={false}
        title={(
            <FormattedMessage
                id='admin.team_settings.team_details.groupDetailsToggle'
                defaultMessage='Enable Group Mention (Beta)'
            />
        )}
        subTitle={(
            <FormattedMarkdownMessage
                id='admin.team_settings.team_details.groupDetailsToggleDescr'
                defaultMessage='When enabled, this group can be mentioned in other channels and teams. This may result in the group member list being visible to all users.'
            />
        )}
    />);

GroupSettingsToggle.propTypes = {
    isDefault: PropTypes.bool.isRequired,
    allowReference: PropTypes.bool.isRequired,
    onToggle: PropTypes.func.isRequired,
    isDisabled: PropTypes.bool,
};

export const GroupProfileAndSettings = ({displayname, mentionname, allowReference, onToggle, onChange, readOnly}) => (
    <AdminPanel
        id='group_profile'
        titleId={t('admin.group_settings.group_detail.groupProfileTitle')}
        titleDefault='Group Profile'
        subtitleId={t('admin.group_settings.group_detail.groupProfileDescription')}
        subtitleDefault='The name for this group.'
    >
        <GroupProfile
            name={displayname}
            title={t('admin.group_settings.group_details.group_profile.name')}
            titleDefault={'Name:'}
            customID={'groupDisplayName'}
            isDisabled={true}
            showAtMention={false}
        />
        <div className='group-settings'>
            <div className='group-settings--body'>
                <div className='section-separator'><hr className='separator__hr'/></div>
                <GroupSettingsToggle
                    isDefault={false}
                    allowReference={allowReference}
                    onToggle={onToggle}
                    isDisabled={readOnly}
                />
            </div>
        </div>
        {allowReference &&
            <GroupProfile
                name={mentionname}
                title={t('admin.group_settings.group_details.group_mention.name')}
                titleDefault={'Group Mention:'}
                customID={'groupMention'}
                isDisabled={readOnly}
                showAtMention={true}
                onChange={onChange}
            />
        }
    </AdminPanel>);

GroupProfileAndSettings.propTypes = {
    displayname: PropTypes.string.isRequired,
    mentionname: PropTypes.string,
    allowReference: PropTypes.bool.isRequired,
    onChange: PropTypes.func.isRequired,
    onToggle: PropTypes.func.isRequired,
    readOnly: PropTypes.bool,
};
