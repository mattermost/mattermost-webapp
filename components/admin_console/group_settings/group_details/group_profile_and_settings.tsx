// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {FormattedMessage} from 'react-intl';

import GroupProfile from 'components/admin_console/group_settings/group_details/group_profile';
import LineSwitch from 'components/admin_console/team_channel_settings/line_switch';
import FormattedMarkdownMessage from 'components/formatted_markdown_message';
import AdminPanel from 'components/widgets/admin_console/admin_panel';

import {t} from 'utils/i18n';

type GroupSettingsToggleProps = {
    isDefault: boolean;
    allowReference: boolean;
    onToggle: (allowReference: boolean) => void;
    isDisabled?: boolean;
};

const GroupSettingsToggle = ({
    isDefault,
    allowReference,
    onToggle,
    isDisabled,
}: GroupSettingsToggleProps) => (
    <LineSwitch
        id={'allowReferenceSwitch'}
        disabled={isDisabled || isDefault}
        toggled={allowReference}
        last={true}
        onToggle={() => {
            if (isDefault) {
                return;
            }
            onToggle(!allowReference);
        }}
        singleLine={false}
        title={
            <FormattedMessage
                id='admin.team_settings.team_details.groupDetailsToggle'
                defaultMessage='Enable Group Mention (Beta)'
            />
        }
        subTitle={
            <FormattedMarkdownMessage
                id='admin.team_settings.team_details.groupDetailsToggleDescr'
                defaultMessage='When enabled, this group can be mentioned in other channels and teams. This may result in the group member list being visible to all users.'
            />
        }
    />
);

type GroupProfileAndSettingsProps = {
    displayname: string;
    mentionname?: string;
    allowReference: boolean;
    onChange: React.ChangeEventHandler<HTMLInputElement>;
    onToggle: (allowReference: boolean) => void;
    readOnly?: boolean;
};

export const GroupProfileAndSettings = ({
    displayname,
    mentionname,
    allowReference,
    onToggle,
    onChange,
    readOnly,
}: GroupProfileAndSettingsProps) => (
    <AdminPanel
        id='group_profile'
        titleId={t('admin.group_settings.group_detail.groupProfileTitle')}
        titleDefault='Group Profile'
        subtitleId={t(
            'admin.group_settings.group_detail.groupProfileDescription',
        )}
        subtitleDefault='The name for this group.'
    >
        <GroupProfile
            name={displayname}
            title={t('admin.group_settings.group_details.group_profile.name')}
            titleDefault={'Name:'}
            customID={'groupDisplayName'}
            isDisabled={true}
            showAtMention={false}
        />
        <div className='group-settings'>
            <div className='group-settings--body'>
                <div className='section-separator'>
                    <hr className='separator__hr'/>
                </div>
                <GroupSettingsToggle
                    isDefault={false}
                    allowReference={allowReference}
                    onToggle={onToggle}
                    isDisabled={readOnly}
                />
            </div>
        </div>
        {allowReference && (
            <GroupProfile
                name={mentionname}
                title={t(
                    'admin.group_settings.group_details.group_mention.name',
                )}
                titleDefault={'Group Mention:'}
                customID={'groupMention'}
                isDisabled={readOnly}
                showAtMention={true}
                onChange={onChange}
            />
        )}
    </AdminPanel>
);
