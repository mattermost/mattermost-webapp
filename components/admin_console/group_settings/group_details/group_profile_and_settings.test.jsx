// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import {GroupProfileAndSettings} from './group_profile_and_settings.jsx';

describe('components/admin_console/group_settings/group_details/GroupProfileAndSettings', () => {
    test('should match snapshot, with toggle off', () => {
        const wrapper = shallow(
            <GroupProfileAndSettings
                displayname='GroupProfileAndSettings'
                mentionname='GroupProfileAndSettings'
                allowReference={false}
                onChange={jest.fn()}
                onToggle={jest.fn()}
            />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, with toggle on', () => {
        const wrapper = shallow(
            <GroupProfileAndSettings
                displayname='GroupProfileAndSettings'
                mentionname='GroupProfileAndSettings'
                allowReference={true}
                onChange={jest.fn()}
                onToggle={jest.fn()}
            />);
        expect(wrapper).toMatchSnapshot();
    });
});