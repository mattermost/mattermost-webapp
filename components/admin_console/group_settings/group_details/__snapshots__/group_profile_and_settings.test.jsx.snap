// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/admin_console/group_settings/group_details/GroupProfileAndSettings should match snapshot, with toggle off 1`] = `
<AdminPanel
  className=""
  id="group_profile"
  subtitleDefault="The name for this group."
  subtitleId="admin.group_settings.group_detail.groupProfileDescription"
  titleDefault="Group Profile"
  titleId="admin.group_settings.group_detail.groupProfileTitle"
>
  <GroupProfile
    customID="groupDisplayName"
    isDisabled={true}
    name="GroupProfileAndSettings"
    showAtMention={false}
    title="admin.group_settings.group_details.group_profile.name"
    titleDefault="Name:"
  />
  <div
    className="group-settings"
  >
    <div
      className="group-settings--body"
    >
      <div
        className="section-separator"
      >
        <hr
          className="separator__hr"
        />
      </div>
      <GroupSettingsToggle
        allowReference={false}
        isDefault={false}
        onToggle={[MockFunction]}
      />
    </div>
  </div>
</AdminPanel>
`;

exports[`components/admin_console/group_settings/group_details/GroupProfileAndSettings should match snapshot, with toggle on 1`] = `
<AdminPanel
  className=""
  id="group_profile"
  subtitleDefault="The name for this group."
  subtitleId="admin.group_settings.group_detail.groupProfileDescription"
  titleDefault="Group Profile"
  titleId="admin.group_settings.group_detail.groupProfileTitle"
>
  <GroupProfile
    customID="groupDisplayName"
    isDisabled={true}
    name="GroupProfileAndSettings"
    showAtMention={false}
    title="admin.group_settings.group_details.group_profile.name"
    titleDefault="Name:"
  />
  <div
    className="group-settings"
  >
    <div
      className="group-settings--body"
    >
      <div
        className="section-separator"
      >
        <hr
          className="separator__hr"
        />
      </div>
      <GroupSettingsToggle
        allowReference={true}
        isDefault={false}
        onToggle={[MockFunction]}
      />
    </div>
  </div>
  <GroupProfile
    customID="groupMention"
    isDisabled={false}
    name="GroupProfileAndSettings"
    onChange={[MockFunction]}
    showAtMention={true}
    title="admin.group_settings.group_details.group_mention.name"
    titleDefault="Group Mention:"
  />
</AdminPanel>
`;
